modules/snooker/src/main/ts/ephox/snooker/api/TableOperations.ts

```diff
@@ -324,28 +346,36 @@ const opPasteRowsAfter = (grid: Structs.RowCells[], pasteDetails: ExtractPasteRo
   return outcome(mergedGrid, cursor);
 };
 
-const opGetColumnType = (table: SugarElement, target: TargetSelection): string => {
+const isTableHeaderCell = SugarNode.isTag('th');
+
+const findCommonCellType = (cells: Structs.DetailExt[]): Optional<string> => {
+  const headerCells = Arr.filter(cells, (cell) => isTableHeaderCell(cell.element));
+  if (headerCells.length === 0) {
+    return Optional.some('td');
+  } else if (headerCells.length === cells.length) {
+    return Optional.some('th');
+  } else {
+    return Optional.none();
+  }
+};
+
+const opGetColumnsType = (table: SugarElement, target: TargetSelection): string => {
   const house = Warehouse.fromTable(table);
   const details = RunOperation.onCells(house, target);
-  return details.bind((selectedCells): Optional<string> => {
+  return details.bind((selectedCells) => {
     const lastSelectedCell = selectedCells[selectedCells.length - 1];
     const minColRange = selectedCells[0].column;
     const maxColRange = lastSelectedCell.column + lastSelectedCell.colspan;
     const selectedColumnCells = Arr.flatten(Arr.map(house.all, (row) =>
       Arr.filter(row.cells, (cell) => cell.column >= minColRange && cell.column < maxColRange)));
-    return getCellsType(selectedColumnCells, (cell) => SugarNode.name(cell.element) === 'th');
+    return findCommonCellType(selectedColumnCells);
   }).getOr('');
 };
 
-export const getCellsType = <T>(cells: T[], headerPred: (x: T) => boolean): Optional<string> => {
-  const headerCells = Arr.filter(cells, headerPred);
-  if (headerCells.length === 0) {
-    return Optional.some('td');
-  } else if (headerCells.length === cells.length) {
-    return Optional.some('th');
-  } else {
-    return Optional.none();
-  }
+const opGetCellsType = (table: SugarElement, target: TargetSelection): string => {
```

Comment:
`op` prefix is not only for `Optional` return type?

Reply:
`op` === `operation` in this case.